src/web/functions.php

```diff
@@ -11,7 +11,93 @@
  */
 function getUniqueFirstLetters(array $airports)
 {
-    // put your logic here
+    $firstLetters = [];
 
-    return ['A', 'B', 'C'];
-}
\ No newline at end of file
+    foreach ($airports as $airport) {
+        $firstLetters[] = $airport['name'][0];
+    }
+
+    sort($firstLetters);
+
+    return array_unique($firstLetters);
+}
+
+function filterByFirstLetter($airports)
+{
+    $airports_by_letter = [];
+
+    foreach ($airports as $key => $airport) {
+        if ($airport['name'][0] == $_GET['filter_by_first_letter']) {
+            $airports_by_letter[] = $airport;
+        }
+    }
+
+    return $airports_by_letter;
+}
+
+function filterByState($airports)
+{
+    $airports_by_state = [];
+
+    foreach ($airports as $key => $airport) {
+        if ($airport['state'][0] == $_GET['filter_by_state']) {
```

Comment:
Please use the function parameter instead of direct using $_GET in all such cases